File: weekly_390th/A.cpp
class Solution {
public:
    int maximumLengthSubstring(string s) {
        vector<int>cnt(26);
        int res = 0;
        int l = 0; int r = 0;
        while(r < s.size()) {
            cnt[s[r] - 'a']++;
            while(cnt[s[r] - 'a'] > 2) {
                cnt[s[l] - 'a']--;
                l++;
            }
            res = max(res, r - l + 1);
            r++;
        }
        return res;
    }
};

File: weekly_390th/A.go
func maximumLengthSubstring(s string) int {
    ans := 0
    cnt := [26]int{}
    l := 0
    r := 0
    n := len(s)
    for r < n {
        cnt[rune(s[r]) - 'a']++
        for cnt[rune(s[r]) - 'a'] > 2 {
            cnt[rune(s[l]) - 'a']--
            l++
        }
        ans = max(ans, r - l + 1);
        r++
    }
    return ans
}
